src/main/daml/Daml/Finance/Claims/Util/Builders.daml

```diff
@@ -207,23 +207,26 @@ createCreditEventPaymentClaims dateToTime ownerReceives notional cashInstrument
 
 -- ASSET_PERFORMANCE_CLAIMS_BEGIN
 -- | Calculate the asset performance for each payment date and create claims.
--- The performance is calculated using the reference asset from the start date to the end date of
--- each payment period. The reference asset Observable needs to contain the appropriate type of
--- fixings:
+-- The performance is calculated using the weighted performance of the basket underlyings from the
+-- start date to the end date of each payment period.
+-- The reference asset Observables need to contain the appropriate type of fixings:
 --   - unadjusted fixings in case of a price return asset swap
 --   - adjusted fixings in case of a total return asset swap
 createAssetPerformancePaymentClaims : (Date -> Time) -> Schedule -> PeriodicSchedule -> Bool ->
-  Bool -> DayCountConventionEnum -> Decimal -> Deliverable -> Observable -> TaggedClaim
+  Bool -> DayCountConventionEnum -> Decimal -> Deliverable -> [(Text, Decimal)]
+  -> TaggedClaim
 createAssetPerformancePaymentClaims dateToTime schedule periodicSchedule useAdjustedDatesForDcf
-  ownerReceives dayCountConvention notional cashInstrument referenceAssetId =
+  ownerReceives dayCountConvention notional cashInstrument underlyingRefsAndWeights =
   let
     assetClaimAmounts = foldMap (\p ->
       let
-        performance =
-          Observe referenceAssetId / ObserveAt referenceAssetId p.adjustedStartDate - Const 1.0
+        calcBasketPerformance acc (referenceAssetId, weight) =
```

Comment:
If I remember correctly `Observation` is a monoid, so you could use `foldMap` which reads a bit simpler

Reply:
got it, so `Observation` is no monoid after all